task_sdk/src/airflow/sdk/definitions/asset/__init__.py

```diff
@@ -429,10 +429,13 @@ class AssetAlias(BaseAsset):
     group: str = attrs.field(kw_only=True, default="asset", validator=_validate_identifier)
 
     def _resolve_assets(self) -> list[Asset]:
+        from airflow import settings
         from airflow.models.asset import expand_alias_to_assets
-        from airflow.utils.session import create_session
 
-        with create_session() as session:
+        # We don't use create_session or provide_session here because this is
+        # called in the scheduler when commit is prohibited (for HA reasons).
+        # Those functions assume you want to commit the result on exit.
```

Comment:
Isn't it possible to pass the scheduler session to this?

Reply:
It is, but this function is also (indirectly) used by places where there’s not an active session. I feel it’s easier to open a new one for this (it doesn’t really cost much).

Reply:
I changed this to accept an optional session when it makes sense instead.